### PR TITLE
Persist photo state and upload assets to s3

### PR DESCRIPTION
**Pull request by orta**
This makes it possible for the Overview to trigger uploads to S3 and Gemini.

Relies on https://github.com/artsy/metaphysics/pull/786

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>

  No CHANGELOG added.
  </td>
    </tr>
  </tbody>
</table>




<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>

  No CHANGELOG added.
  </td>
    </tr>
  
<tr>
      <td>:no_entry_sign:</td>
      <td>

  Missing Test Files:

- `src/lib/Components/Consignments/Submission/__tests__/uploadPhotoToGemini-tests.ts`

If these files are supposed to not exist, please update your PR body to include "Skip New Tests".
  </td>
    </tr>
  </tbody>
</table>




## TSLint Issues:

* :no_entry_sign: <a href="https://github.com/artsy/emission/blob/photos/src/lib/Components/Consignments/Submission/uploadPhotoToGemini.ts">uploadPhotoToGemini.ts</a> - no-debugger - Use of debugger statements is forbidden

<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

File: package.json [hunk 1]
```diff
@@ -59,6 +59,7 @@
     "lodash": "^4.17.4",
     "moment": "latest",
     "prop-types": "^15.5.10",
+    "query-string": "^4.1.0",
```
Comment (orta) [inline package.json]: Already in the dep tree, but now it is explicitly used.

File: src/lib/Components/Consignments/Components/ImageSelection.tsx [hunk 2]
```diff
@@ -96,7 +98,7 @@ export default class ImageSelection extends React.Component<Props, State> {
     super(props)
 
     this.state = {
-      selected: new Map(),
+      selected: props.selected ? props.selected : new Set(),
```
Comment (orta) [inline src/lib/Components/Consignments/Components/ImageSelection.tsx]: Switched to a set as it's easier to manipulate values

File: src/lib/Components/Consignments/Components/ArtworkConsignmentTodo.tsx [hunk 3]
```diff
@@ -92,7 +92,7 @@ const Button: React.SFC<TouchableHighlightProperties> = ({ children, ...props })
   </TouchableHighlight>
 
 const ImagePreview = images =>
-  <ImageStyle source={{ uri: images[0] }}>
+  <ImageStyle source={{ uri: images[0].file }}>
```
Comment (orta) [inline src/lib/Components/Consignments/Components/ArtworkConsignmentTodo.tsx]: needed to let photos store state about whether they've been added or not yet

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>

  Missing Test Files:

- `src/lib/Components/Consignments/Submission/__tests__/uploadPhotoToGemini-tests.ts`

If these files are supposed to not exist, please update your PR body to include "Skip New Tests".
  </td>
    </tr>
  </tbody>
</table>




<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

File: src/lib/Components/Consignments/Screens/__tests__/Overview-local-storage-tests.tsx [hunk 4]
```diff
@@ -42,8 +42,9 @@ it("updates the local state when there an update is triggered", () => {
   )
 })
 
-it("resets the cache when a final submission is made", async () => {
+it("resets the cache when a final submission is made ", async () => {
```
Comment (ashfurrow) [inline src/lib/Components/Consignments/Screens/__tests__/Overview-local-storage-tests.tsx]: nitpick: the space added here.

**Pull request merged**